sphinx/ext/graphviz.py

```diff
@@ -261,7 +261,7 @@ def render_dot(self, code, options, format, prefix='graphviz'):
         # Graphviz may close standard input when an error occurs,
         # resulting in a broken pipe on communicate()
         stdout, stderr = p.communicate(code.encode('utf-8'))
-    except (OSError, IOError) as err:
+    except OSError as err:
```

Comment:
Given the `errno` check below, could this catch `BrokenPipeError` instead?

https://docs.python.org/3/library/exceptions.html#BrokenPipeError

Reply:
Yeah, that is right. So `OSError` is still necessary to handle the `EINVAL` case, so there isn't much point to changing it. Do you know what triggers an `EINVAL`? I tried to find more details, but didn't find anything specific.

Reply:
It comes from 9350b3b46b. The commit message says:

>This happen on Python2.7 and Graphviz version 2.12 (Mon Dec  4 22:04:37 UTC 2006)
on Windows when a not recognized renderer type is asked.

I don't know latest python and windows causes the `EINVAL` error. We have to confirm that before removing...

Reply:
The `EINVAL` has not been thrown at present. That was fixed at https://bugs.python.org/issue19612
https://github.com/python/cpython/blob/master/Lib/subprocess.py#L880

@gjo Thank you for advice!